uc.c

```diff
@@ -2418,6 +2431,11 @@ uc_err uc_context_restore(uc_engine *uc, uc_context *context)
             return ret;
         }
         uc_snapshot(uc);
+        uc->ram_list.freed = context->ramblock_freed;
```

Comment:
Sorry I should have noticed this earlier. With this change (and actually the early `snapshot_level` change), with `UC_CTL_CONTEXT_MEMORY`, the context is no longer safe to keep as a file (it contains pointers) and restored to another instance.

It is super non-trivial to adapt the memory context to a value type and thus I suggest adding a few documents in the header (`uc_context_save/restore`, `UC_CTL_CONTEXT_MEMORY`).

Reply:
Yes, we had this question already in the snapshots feature. I'll write some documentation about this.